draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -831,16 +834,82 @@ Instantiating one of these protocols with an elliptic curve group motivates
 the term "hashing to the curve", i.e., encoding bitstrings to points on an
 elliptic curve.
 
-The hash2curve(alpha) function can be easily constructed by
-using the output of a cryptographically secure hash function H as the input of
-an encoding function.
-On the one hand, hash2curve is difficult to invert since it is computationally
-intractable to produce an input alpha that maps to hash2curve(alpha) due to H
-is pre-image resistant.
+One can easily construct a hash to curve function h(x) by using the output of a
+cryptographically secure hash function H as the input of an encoding function.
+On the one hand, h(x) is difficult to invert since it is computationally
+intractable to produce an input x that maps to h(x) due to H is pre-image
+resistant.
 On the other hand, the uniformity property is not met as the output of an
 encoding is distinguishable from a random distribution. Hence, using
-hash2curve(alpha) is not sufficient to get uniformity, however it can be used
-as a building block for obtaining a random oracle as is described in {{rom}}.
+h(x) is not sufficient to get uniformity and cannot be used as a random oracle.
+
+Brier et al. {{BCIMRT10}} describe two generic constructions that give a
+hash function approximating a random oracle. Farashahi et al. {{FFSTV13}}
+and Tibouchi and Kim {{TK17}} refine this analysis. In particular, Farashahi
+et al. show that summing two independent evaluations of any of the
+deterministic endings of {{encodings}} suffices to approximate a random
+oracle to an elliptic curve. In other words:
+
+~~~
+   h(x) = f(H0(x)) + f(H1(x))
+~~~
+
+where f: F -> E(F) is a deterministic encoding, H0 and H1 are hash
+functions (modeled as random oracles) mapping strings to elements of F, and the
+addition operation corresponds to point addition on the target elliptic curve.
+
+
+# Roadmap on Hash to Curve. {#roadmap}
+
+The following functions provide a general framework for mapping bit
+strings into points of a prime group G.
```

Comment:
We should expand on this a bit and more clearly highlight the differences between encoding and hashing to the curve. Perhaps:

```
A mapping can either yield a uniform distribution over the range of points on the curve or not. A mapping that does not yield a uniform distribution is an injective encoding. Example algorithms include Elligator2, wherein the algorithm maps input strings to approximately half of the points on the curve. A mapping that yields a uniform distribution is, effectively, a random oracle from strings to points on the curve. Where appropriate, we use the terms random oracle and hash functions for such mappings interchangeably.```

Reply:
Also, to Chris's point: the first paragraph of the section should introduce (or re-introduce) the high-level concepts before jumping into the low-level details.

Maybe the right way to do this is to add a new paragraph before what's currently here that tells the reader what's ahead:

- high level: three pieces (hash2base, map2curve, clear_h)
- the following sections define each piece in detail
- {{term-rom}} discussed two possibilities: injective mapping vs random oracle

etc